server/odc-service/src/main/java/com/oceanbase/odc/service/db/DBPLModifyHelper.java

```diff
@@ -0,0 +1,167 @@
+                                    + OB_VERSION_SUPPORT_MULTIPLE_SAME_TRIGGERS);
+                }
+                return executeWrappedEditPL(sessionId, editPLReq, DBObjectType.TRIGGER, ODC_TEMPORARY_TRIGGER);
+            default:
+                throw new IllegalArgumentException(
+                        String.format("the pl type %s of editing procedure is not supported", plType));
+        }
+    }
+
+    private EditPLResp executeWrappedEditPL(String sessionId, EditPLReq editPLReq,
+            DBObjectType plType, String tempPlName) throws Exception {
+        String plName = editPLReq.getObjectName();
+        String editPLSql = editPLReq.getSql();
+        String tempPLSql = editPLSql.replaceFirst(plName, tempPlName);
+        StringBuilder wrappedSqlBuilder = new StringBuilder();
+        wrappedSqlBuilder.append("DELIMITER $$\n")
+                .append(tempPLSql).append(" $$\n")
+                .append("drop ").append(plType).append(" if exists ").append(tempPlName).append(" $$\n")
+                .append("drop ").append(plType).append(" if exists ").append(plName).append(" $$\n")
+                .append(editPLSql).append(" $$\n")
+                .append("DELIMITER ;");
+        String wrappedSql = wrappedSqlBuilder.toString();
+        SqlAsyncExecuteReq sqlAsyncExecuteReq = new SqlAsyncExecuteReq();
+        sqlAsyncExecuteReq.setSql(wrappedSql);
+        sqlAsyncExecuteReq.setSplit(true);
+        sqlAsyncExecuteReq.setContinueExecutionOnError(false);
+        ConnectionSession connectionSession = sessionService.nullSafeGet(sessionId, true);
+        Lock editPLLock = obtainEditPLLock(connectionSession, plType);
+        if (!editPLLock.tryLock(LOCK_TIMEOUT_SECONDS, TimeUnit.SECONDS)) {
+            throw new ConflictException(ErrorCodes.ResourceModifying, "Can not acquire jdbc lock");
+        }
+        try {
+            SqlAsyncExecuteResp sqlAsyncExecuteResp = connectConsoleService.streamExecute(sessionId, sqlAsyncExecuteReq,
+                    true);
+            EditPLResp editPLResp = new EditPLResp();
+            editPLResp.setWrappedSql(wrappedSql);
+            editPLResp.setSqls(sqlAsyncExecuteResp.getSqls());
+            editPLResp.setUnauthorizedDBResources(sqlAsyncExecuteResp.getUnauthorizedDBResources());
+            editPLResp.setViolatedRules(sqlAsyncExecuteResp.getViolatedRules());
+            editPLResp.setApprovalRequired(sqlAsyncExecuteResp.isApprovalRequired());
+            if (editPLResp.isApprovalRequired()) {
+                return editPLResp;
+            }
+            AsyncExecuteResultResp moreResults;
+            List<SqlExecuteResult> results = new ArrayList<>();
+            do {
+                moreResults = connectConsoleService.getMoreResults(sessionId,
+                        sqlAsyncExecuteResp.getRequestId());
+                results.addAll(moreResults.getResults());
+            } while (!moreResults.isFinished());
+            for (SqlExecuteResult result : results) {
```

Comment:
how about add these `List<SqlExecuteResult> results` into your EditPLResp? the front end can reuse the old logic.

btw, you can ignore this comment if the front end accept EditPLResp

Reply:
ok